### DISCUSSION TITLE
How to get all messages from chat history.

### DISCUSSION THREAD

**Discussion by ainizoda** [gram-js/gramjs#358] [Q&A] [answered]
I'm trying to get and it always returns 100 messages, no more

**Comment by rojvv** [comment, marked as answer]
You can do that by iterating through them:

```ts
for await (const message of client.iterMessages(yourChat)) {
  // use message
}
```

This is how you can use the above method to put all of them into an array:

```ts
const messages = new Array<Api.Message>();
for await (const message of client.iterMessages(yourChat)) {
  messages.push(message);
} 
```

**Reply by ainizoda** [reply]
Thank you, finally it works, but how can I do that for channels (groups)?

**Reply by rojvv** [reply]
Just pass its ID to `iterMessages`?

**Reply by ainizoda** [reply]
Nope, it doesn't work.

**Reply by painor** [reply]
can you show your code and the error you get?

**Reply by privorka** [reply]
Don't forget, that it needs the `user authorisation`, it cannot be called using `bot token` for the authorisation as telegram history is not accessible for bots.
